crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -470,6 +470,17 @@ define_tedge_config! {
         #[doku(as = "PathBuf")]
         root_cert_path: Utf8PathBuf,
 
+        // TODO: Enum or bool? If enum, what are the other items?
+        /// Use basic authentication (username/password) instead of device certificate based authentication
+        #[tedge_config(example = "true", default(value = false))]
+        use_basic_auth: bool,
```

Comment:
> If device credentials are configure for c8y, then uses HTTP basic-auth to connect c8y. Otherwise, use a JWT token.

@reubenmiller 
Can you give me UX perspective here? For me, both (credentials are configured or enum) sound good...

Reply:
@rina23q I would go with an Enum as there is a case if someone wants to transition from username/password to token based auth (via certificates), then you'd change the mode to JWT, verify that the new connection works, then delete the old username/password file.

But in some cases I do like the auto detection stuff, so maybe we can do both, but having another enum value:

* certificate (e.g. jwt)
* basic
* auto (will use basic if username/password are provided)

Reply:
Implemented in 4e41c4efd320286ba796c87d57122dbe69077d13